pkg/cmd/rollbackcopy/fileioutils.go

```
@@ -0,0 +1,149 @@
+package rollbackcopy
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"k8s.io/klog"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+func dirExists(dirname string) bool {
+	info, err := os.Stat(dirname)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return info.IsDir()
+}
+
+func safeDirRename(src, dest string, srcMayNotExist bool) error {
+	if !dirExists(src) {
+		if !srcMayNotExist {
+			return fmt.Errorf("SafeDirRename: src dir %s does not exist", src)
+		} else {
+			return nil
+		}
+	}
+	destExists := false
+	destToBeRemoved := dest + ".to_be_removed"
+	if dirExists(dest) {
+		destExists = true
+		if dirExists(destToBeRemoved) {
+			os.RemoveAll(destToBeRemoved)
+		}
+		if err := os.Rename(dest, destToBeRemoved); err != nil {
+			klog.Error("Got error ", err)
+			return err
+		}
+	}
+
+	if err := os.Rename(src, dest); err != nil {
+		if destExists {
+			os.Rename(destToBeRemoved, dest)
+		}
+		klog.Error("Got error ", err)
```

Wrap and return error instead of logging